hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/MergeOnReadSnapshotRelation.scala

```diff
@@ -61,6 +61,11 @@ class MergeOnReadSnapshotRelation(sqlContext: SQLContext,
                                     tableSchema: HoodieTableSchema,
                                     requiredSchema: HoodieTableSchema,
                                     filters: Array[Filter]): HoodieMergeOnReadRDD = {
+    if (!internalSchema.isEmptySchema) {
+      // it is safe to enable vectorizedReader
+      sqlContext.sparkSession.sessionState.conf.setConfString("spark.sql.parquet.recordLevelFilter.enabled", "false")
```

Comment:
@xiarixiaoyao : Also, here

Reply:
we no  need  recordLevelFilter.enabled  for  snapshot read,    this filter only for incremental query，  mor incremental query need to filter data through file level filters
i also  post another pr to  fixed  regression problem https://github.com/apache/hudi/pull/5168
once  merged  will remove  those code.